app/bundles/PluginBundle/Form/Type/FeatureSettingsType.php

```diff
@@ -66,11 +66,29 @@ public function buildForm (FormBuilderInterface $builder, array $options)
 
             list ($specialInstructions, $alertType) = $integration_object->getFormNotes('leadfield_match');
 
+            /**
+             * Auto Match Integration Fields with Mautic Fields.
+             */
+            $flattenLeadFields = array();
+            foreach (array_values($leadFields) as $fieldsWithoutGroups) {
+                $flattenLeadFields = array_merge($flattenLeadFields,$fieldsWithoutGroups);
+            }
+            
+            $integrationFields  = array_keys($fields);
+            $flattenLeadFields  = array_keys($flattenLeadFields);
+            $fieldsIntersection = array_uintersect($integrationFields, $flattenLeadFields, "strcasecmp");
+
+
+            $autoMatchedFields = array();
+            foreach ($fieldsIntersection as $field) {
+              $autoMatchedFields[$field] = $field;
```

Comment:
I've fixed the code, and if you `$data['leadFields']`is returning
```
array(2) {
  ["Company"]=> string(9) "firstname"
  ["LastName"]=> string(8) "lastname"
}
```
Check if these settings are not the ones that you have saved at the database, because it will only auto-match if no settings come from the DB. So it'll only work for the first configuration.